_posts/2022-07-27-osbuild-operator-for-os-lifecycle-in-flotta.markdown

```diff
@@ -0,0 +1,206 @@
+```shell
+oc create secret generic osbuild-s3-credentials -n osbuild --from-literal=access-key-id=minioadmin --from-literal=secret-access-key=minioadmin
+
+```
+Create a secret for the CA Bundle using the OCP route
+```shell
+oc get secrets -n openshift-ingress-operator router-ca -o "jsonpath={.data.tls\.crt}" | base64 -d > /tmp/ca-bundle
+oc create secret generic osbuild-s3-ca-bundle -n osbuild --from-file=/tmp/ca-bundle
+
+```
+
+### Create a Secret for your Red Hat Credentials
+
+Find your Red Hat credentials and create a secret:
+```shell
+oc create secret generic redhat-portal-credentials -n osbuild --from-literal=username=<USERNAME> --from-literal=password=<PASSWORD>
+```
+
+### Create a PostgreSQL Server
+Currently, the controller does not support creating the PostgreSQL server on its own, making this step mandatory
+
+Edit the file `config/creating_env/psql.env` and set a real password
+
+Create the PostgreSQL server by running 
+```shell
+oc new-app --env-file config/creating_env/psql.env postgresql:13-el8 -n osbuild
+```
+
+Edit the file `config/creating_env/postgress_secret.yaml` with the same password
+
+Create new secret (please enter a real encoded password) 
+```shell
+oc create -f config/creating_env/postgress_secret.yaml
+```
+
+### Provision an External Worker VM using CNV
+Create an SSH key-pair
+```shell
+ssh-keygen -t rsa -b 4096 -C cloud-user@external-builder -f ~/.ssh/external-builder
+```
+
+Create symlinks to the files to facilitate the next step
+```shell
+ln -s ~/.ssh/external-builder.pub config/creating_env/ssh-publickey
+ln -s ~/.ssh/external-builder config/creating_env/ssh-privatekey
+```
+
+Generate the secret
+```shell
+oc create secret generic external-builder-ssh-pair --from-file=config/creating_env/ssh-privatekey --from-file=config/creating_env/ssh-publickey -n osbuild
+```
+
+Deploy the VM
+```shell
+oc apply -n osbuild -f config/creating_env/external-worker-vm.yaml
+```
+
+Get VM Address
+```shell
+oc get vmi external-builder -o jsonpath={.status.interfaces[0].ipAddress}
+```
+
+
+## Create OSBuildEnvConfig singleton CR ##
+Apply OSBuildEnvConfig
+```shell
+export CLUSTER_DOMAIN='mycluster.example.com'
+export EXTERNAL_WORKER_IP=`oc get vmi external-builder -o jsonpath={.status.interfaces[0].ipAddress}`
+cat config/samples/osbuilder_v1alpha1_osbuildenvconfig.yaml | envsubst | oc apply -f -
+```
+
+## Create an edge-container image ##
+Apply osbuildConfig that contains all the relevant configuration for running osbuild-composer job.
+You can use that sample:
+```shell
+oc create -f config/samples/osbuilder_v1alpha1_osbuildconfig.yaml -n osbuild
+```
+
+After applying osbuildConfig CR you should see a new osbuild CR that was triggered automatically
+```shell
+oc describe osbuild osbuildconfig-sample-1 -n osbuild 
+```
+
+As part of that osbuild CR you can found the osbuild-composer job status, the job composer ID, and after finish running you also can found the containerUrl in the S3 bucket
```

Comment:
found - > find
and after finish running you also can find -> and once the job finishes, you can also find